app/controllers/course/leaderboards_controller.rb

```diff
@@ -3,12 +3,17 @@ class Course::LeaderboardsController < Course::ComponentController
   before_action :check_component
   before_action :load_settings
   before_action :add_leaderboard_breadcrumb
+  before_action :check_group_leaderboard_settings, only: [:groups]
```

Comment:
Not sure if that is a good idea cause `check_component` checks the existence of the component. I don't think we want to add the logic to check the settings of that component inside that method as well.

Maybe we can just have a `check_component_settings` that uses a case statement instead?